### ISSUE TITLE
Training Stuck on Resizing Images

### ISSUE THREAD

**Issue by LouisMach** [experiencor/keras-yolo3#271]
I am using the SVHN dataset to train a digit detection model.
This dataset contains approximately 33 000 pictures, most of which are 110x60 pixels.
When I first tried to train, I ran into Out Of Memory errors by Tensorflow. The batch size of 16 seemed to be too much for the 1080ti which has 11GB of VRAM. I also got this error with a batch size of 10.
Finally, I changed the batch size to 4. After this, I no longer get an OOM error.
Using the train.py with a the modified config file, I get some tensorflow output which seems to indicate training is starting as normal. However, the program starts to output "resizing: " and doesn't seem to progress in epochs. I have left it training for over 2 hours and all I've seen so far is hundreds of lines of resizing operations.
![image](https://user-images.githubusercontent.com/36770076/81433211-7c83f000-9164-11ea-8cfe-392e9421f353.png)

Is this to be expected? Should the model be trained eventually? How long should training take for 33k 110x60 images on a 1080ti? Could I reduce the training set to solve this problem and still have a decent model?

**Comment by zmlim** [comment]
hi @Flozker did you manage to solve the issue? I'm currently facing the same problem as well

**Comment by LouisMach** [comment]
@zmlim from what I've found so far, it's expected. There's just a lot of resizing between epochs, and training takes longer than expected for me. I ended up training with a smaller dataset, 1200 pictures instead of 33000. So far results could be better, so I might try increasing the dataset again.

I timed the resizing operation to see how much time was being taken by resizing, but it's a miniscule amount of time per operation. According to perfcounter it's about 1.1000000057492798e-06 seconds. Resizing happens every 10 batches so for my dataset of 30k pictures with a batch size of 8 and 100 epochs, that means 37 500 resizes which takes 0.04 seconds in total. Unfortunately, training seems to take a long time even on the 1080ti regardless of resizing.

**Comment by zmlim** [comment]
hi @Flozker thank you for the timely reply! may I know how many classes are you training on and the respective mAP? My mAP is really low at 0.3184, training on 3 classes, total of 3635 images

**Comment by LouisMach** [comment]
@zmlim 
I just finished training the 10 classes from the SVHN dataset using 3000 training images. According to the validation I get an mAP of 0.8756
![image](https://user-images.githubusercontent.com/36770076/81682041-e3e7bb80-9454-11ea-8120-b46315a1b04e.png)
Weirdly though, when I try to use this model I get mostly incorrect results:
![image](https://user-images.githubusercontent.com/36770076/81682476-25786680-9455-11ea-9b82-eebde5a10e20.png)

**Comment by zmlim** [comment]
@Flozker I tried retraining my dataset with a higher batch size (from 16 to 32) and the training time is even lower. 8 hours and barely done with 2 epoch.  Do you have any idea what is the reason for the resizing ?

**Comment by LouisMach** [comment]
@zmlim from what I understand, resizing the input of the neural network improves the accuracy of the resulting model.

**Comment by zmlim** [comment]
@Flozker Add ```labels.sort()``` in the bbox.py in the utils folder. I think that might be causing the incorrect results! Because for training of images, the code actually sorted the labels. But for predicting of validation dataset, the labels was not sorted, hence, it might be the reason why your results are inaccurate. 

```
def draw_boxes(image, boxes, labels, obj_thresh, quiet=True):
    for box in boxes:
        label_str = ''
        label = -1
        labels.sort()
```

**Comment by LouisMach** [comment]
Thanks for the suggestion! Unfortunately, my results are still far from useable.
The model is mostly correct on this image from the training set:
![python_2020-05-27_15-53-11](https://user-images.githubusercontent.com/36770076/83029411-f22bff00-a032-11ea-8612-5ab3d3904d01.png)
But on another image the bounding boxes are situated inside the numbers instead of around them.
![python_2020-05-27_15-52-43](https://user-images.githubusercontent.com/36770076/83029454-feb05780-a032-11ea-9b3c-9940b7893a87.png)
